src/main.rs

```diff
@@ -87,13 +76,11 @@ fn cache_visit(cache: Store) -> impl warp::Reply {
     let mut timed_string = cache.write().unwrap();
     // check if someone else acquired the write lock while we were blocked
     // and updated the cache already, so just return result
-    if Instant::now() < timed_string.time {
+    if timed_string.is_valid() {
         return Response::new(timed_string.text.clone());
     }
 
     // otherwise we have to update the cache, slowest case.
-    timed_string.time = Instant::now() + Duration::from_secs(10);
-    let resp_text = reqwest::get(&timed_string.url).unwrap().text().unwrap();
-    timed_string.text = Bytes::from(resp_text);
+    timed_string.refresh();
```

Comment:
moved this logic into a shared function as well, this is the default behavior you get from a `TimedString::new()`